src/Types/Prelude.hs

```diff
@@ -115,6 +115,9 @@ infixr 9 <...>
 sortOn :: Ord b => Getter a b -> [a] -> [a]
 sortOn l = sortBy (compare `on` view l)
 
+reverseSortOn :: Ord b => Getter a b -> [a] -> [a]
+reverseSortOn l = sortBy ((flip compare) `on` view l)
```

Comment:
This is the same as `downSortOn`, except it doesn't crash ghc.  I would like to use `downSortOn` instead of this new function and rewrite `downSortOn` accordingly if that fixes the problem.

We should confirm that the ghc crash was caused by that function, which I find surprising.